doc/guides/west/build-flash-debug.rst

```diff
@@ -53,86 +53,124 @@ Building: ``west build``
+positional argument::
 
   west build -b <BOARD> path/to/source/directory
 
-Additionally you can specify the build system target using the ``--target``
-(or ``-t``) option. For example, to run the ``clean`` target::
+To specify the build system target to run, use ``--target`` (or ``-t``). For
+example, to run the ``pristine`` target, which deletes all the files in the
+build directory::
 
-  west build -t clean
+  west build -t pristine
 
-You can list all targets with ``ninja help`` (or ``west build -t help``) inside
-the build folder.
+To list all build system targets::
 
-A clean build can be triggered by using the ``--pristine`` (or ``-p``) option.
-This is particularly handy if you want to switch source dirs or boards without
-using a different build dir::
+  west build -t help
+
+To have ``west build`` run the ``pristine`` target before re-running CMake, use
+``--pristine`` (or ``-p``).  For example, to switch board and application,
+which requires a pristine build directory::
 
   west build -b qemu_x86 samples/philosophers
   west build -p -b reel_board samples/hello_world
 
-If you are unsure about whether the command-line parameters you supply to
-``west build`` require a clean build you can let west decide for you by using
-the ``auto`` setting in the ``--pristine`` option::
+To let west decide for you if a pristine build is needed, use ``-p auto``::
 
   west build -p auto -b reel_board samples/hello_world
 
-Finally, you can add additional arguments to the CMake invocation performed by
-``west build`` by supplying additional parameters (after a ``--``) to the
-command. For example, to use the Unix Makefiles CMake generator instead of
-Ninja (which ``west build`` uses by default), run::
+.. tip::
+
+   You can run ``west config build.pristine auto`` to make this setting
+   permanent.
+
+To add additional arguments to the CMake invocation performed by ``west
+build``, pass them after a ``--`` at the end of the command line. For example,
+to use the Unix Makefiles CMake generator instead of Ninja (which ``west
+build`` uses by default), run::
 
-  west build -b reel_board samples/hello_world -- -G'Unix Makefiles'
+  west build -b reel_board -- -G'Unix Makefiles'
 
-As another example, and assuming you have already built a sample, the following
-command adds the ``file.conf`` Kconfig fragment to the files which are merged
-into your final build configuration::
+As another example, to merge the :file:`file.conf` Kconfig fragment into your
+build's :file:`.config`::
 
   west build -- -DOVERLAY_CONFIG=file.conf
 
 Passing additional CMake arguments like this forces ``west build`` to re-run
-CMake, even if a build system has already been generated. You can also force
-a CMake re-run using the ``-c`` (or ``--cmake``) option::
+CMake, even if a build system has already been generated.
+
+To force a CMake re-run, use the ``--cmake`` (or ``--c``) option::
 
   west build -c
 
+Configuration Options
+=====================
```

Comment:
Why move this here? I thought a centralized source for all configuration options was a good idea actually. It gave a clear overview of all the ways you can configure west in. Maybe create a second table for extension commands in `config.rst` and link to it from here?

Reply:
For one thing, the other page is about "core" west, and this is an extension. For another, as I said in the commit message, i think this will be a lot easier for people to find.